### PR TITLE
Feature/stat tracker ag

### PR DESCRIPTION
**Pull request by algibson1**
Add multiple stat tracker methods, add extra dummy data when needed

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -63,5 +73,126 @@ def percentage_ties
...
+  def count_of_teams
+    @teams.count
+  end
+
+  def group_teams
+    @grouped_game_teams = @game_teams.group_by do |team|
+      team.team_id
+    end
+  end
+  #move into helper class?
```
Comment (dani-wilson) [inline lib/stat_tracker.rb:110]: I like your use of notes.

File: spec/stat_tracker_spec.rb [hunk 2]
```diff
@@ -38,4 +38,24 @@
   it "can calculate the pecentage of games that end up in a tie" do
     expect(@stat_tracker_1.percentage_ties).to eq(0.08)
   end
+
+  it "can count games in each season" do
+    expect(@stat_tracker_1.count_of_games_by_season).to eq({"20122013"=>11, "20132014"=>2})
+  end
+
+  #average_goals_per_game
```
Comment (GVarenhorst) [inline spec/stat_tracker_spec.rb]: I love how we went through this process as a sync session. We were able to see the progression of the tests in real time and contribute together.

**Review by dani-wilson — APPROVED** [review]
Everything looks great and nice and neat! I like the commented out notes to go back to later, and I like the helper methods we've already created here

File: spec/stat_tracker_spec.rb [hunk 3]
```diff
@@ -12,30 +13,101 @@
...
+
+  it "can count average goals by season" do
+    expect(@stat_tracker.average_goals_by_season).to eq({"20122013"=>2.05, "20132014"=>2.0})
+  end
+
+  it "can count all teams" do
+    expect(@stat_tracker.count_of_teams).to eq(16)
+  end
+
+  #test offense helper method? Move helper into other class?
```
Comment (algibson1) [inline spec/stat_tracker_spec.rb:60]: We'll need to add tests for all our helper methods later

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -121,7 +126,22 @@ def best_offense
...
+    worst_team_array = offense_helper.min_by {|team, score| score}
+    worst_team_id = worst_team_array[0].to_s
+    worst_team = @teams.find {|team| team.id == worst_team_id }
+    worst_team.team_name
+  end
+
+  def highest_scoring_visitor
+    @grouped_game_teams
+    scores_when_visitor = {}
+    @grouped_game_teams.each do |team, games|
```
Comment (GVarenhorst) [inline lib/stat_tracker.rb:137]: This one was a doozy. The choice break this up into helper methods was a good decision.

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -132,18 +132,43 @@ def worst_offense
     worst_team.team_name
   end
 
-  def highest_scoring_visitor
-    @grouped_game_teams
-    scores_when_visitor = {}
+  def scoring_helper_method(home_or_away_status)
```
Comment (GVarenhorst) [inline lib/stat_tracker.rb:135]: This helper method is incredible.  Our following methods were much DRYer because of this method.

**Review by GVarenhorst — APPROVED** [review]
I don't see any required changes thus far, the only refactoring I can foresee is perhaps looking for anything listed in our initialize for classes we didn't use and removing them if they weren't required in methods or testing.

**Pull request merged**